src/index.js

```diff
@@ -0,0 +1,42 @@
+const fsp = require('fs').promises;
+const { readDataFiles } = require('./utils/data-parser');
+const { MatterParser } = require('./utils/matter-parser');
+
+async function getFileTimes(fileName, addCreated, addModified) {
+  const result = {};
+  if (!addCreated && !addModified) {
+    return result;
+  }
+  const stats = await fsp.stat(fileName);
+  if (addCreated) {
+    result.created = stats.birthtime.toISOString();
+  }
+  if (addModified) {
+    result.modified = stats.mtime.toISOString();
+  }
+  return result;
+}
+
+async function processFrontmatterFiles({
+  inputFilePatterns,
+  dataFilePatterns,
+  data = {},
+  addCreated = false,
+  addModified = false,
+}) {
+  const dataContents = await readDataFiles(dataFilePatterns);
+  const fileData = Object.assign.apply(this, [{}, ...dataContents]);
+  const inputContents = await MatterParser.fromFilePatterns(inputFilePatterns);
+  await Promise.all(
```

Comment:
Shall the promise be returned?

Reply:
The return can be omitted (also the second return inside the Promise.all expression), as the promise of `writeFile` will be resolved with no arguments on success. 

https://nodejs.org/dist/latest-v10.x/docs/api/fs.html#fs_filehandle_writefile_data_options

Reply:
Though one thing is still missing: in the cli.js it should be wrapped into a try/catch expression for error handling